### PR TITLE
Add hometown map visualization for FGLI Issue

### PR DESCRIPTION
**Pull request by tiffanyl9**
## What does this PR do?
It adds the folder 2024 and within that, the folder fgli-map which contains a map with pins visualization of the hometowns of a sample of students separated into those who identify and do not identify as FGLI.

## How was the functionality tested and verified?
HTML file of the map was opened in a browser.

All steps should be completed in the order presented, and you should not move on to the next step until the previous is completed.
- [x] The title of the `html` page is the same as the chart.
- [x] The `meta_data.json` file is filled in correctly (the description is a complete sentence, the title is the same as the visual).
- [x] The chart and its associated pages have been minified using the `npm run process-visual` command.
- [x] The visual is styled similar to other charts (centered title/subtitle, no legends unless absolutely necessary, etc.)
- [x] This visual was approved by the lead developer.
- [x] The visual was approved by the story's stakeholders.
Screen Shot 2024-01-30 at 11 17 22 AM

### CODE REVIEW

**Comment by michplunkett** [comment]
Base level things:
- You need to run the `npm run lint` command to format your code.
- I don't think it's worth including the non-US students on the map as there's no place to put them.
- You could then update the title to state that it's just US students.
- Try putting the legend in a small container that's close to the lower 48 map.
- Make the title more specific. When I see the title I presume this means ALL students, since non-FGLI and FGLI would be a way you could divide the whole student body.

Overall, great work! I'm excited to see this come together and this is a big venture for your first visual. I see the ambition!

**Comment by Nik4002** [comment]
Hi Tiffany! Another quick change: now that I see the difference in colors more clearly, I think it might be a good idea to put FGLI students above non-FGLI students if we're trying to highlight them. I can see that there's at least one FGLI student from the DC area, but they're covered up by the other pins. To fix this, you probably just need to put the FGLI data series after the non-FGLI one in the code.

File: .eslintrc.json [hunk 1]
```diff
@@ -9,6 +9,6 @@
         "es6": true
     },
     "parserOptions": {
-        "sourceType": "module"
+        "ecmaVersion": 8
```
Comment (michplunkett) [inline .eslintrc.json:12]: This is what was causing your error: https://stackoverflow.com/a/53883984

I accidentally left the 'module' part in there.

File: 2024/fgli-map/main.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
+async function getTopolgy() {
```
Comment (michplunkett) [inline 2024/fgli-map/main.js:1]: I added the `getTopology` part to its own function for the sake of clarity.

File: 2024/fgli-map/main.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                        fontSize: 7
+                    },
+                    y: -23
+                }
+            }
+        },
+
+        tooltip: {
+            headerFormat:
+                '<span style="color:{point.color}">\u25CF</span> {point.key}<br/>',
```
Comment (michplunkett) [inline 2024/fgli-map/main.js]: Use the actual character, not the unicode characters.

Comment (michplunkett) [reply 2024/fgli-map/main.js]: https://www.fileformat.info/info/unicode/char/25cf/index.htm

File: 2024/fgli-map/main.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                        fontSize: 7
+                    },
+                    y: -23
+                }
+            }
+        },
+
+        tooltip: {
+            headerFormat:
+                '<span style="color:{point.color}">BLACK CIRCLE</span> {point.key}<br/>',
```
Comment (michplunkett) [inline 2024/fgli-map/main.js]: `BLACK CIRCLE` should be this ⚫ .

**Comment by michplunkett** [comment]
General notes:
- Standardize legend location
- Standardize zoom button location and potentially see if there's a way to get full zoom out
- Potential title: `Hometown of Surveyed Students from the United States`
- Place ONLY unique cities

**Comment by michplunkett** [comment]
Could you update the image in the description?

**Review by michplunkett — APPROVED** [review]
Excellent work! Well done, @tiffanyl9!

**Pull request merged**